docs/internal/DistributedArchitectureGuide.md

```diff
@@ -12,6 +12,214 @@ A guide to the general Elasticsearch components can be found [here](https://gith
 
 # Networking
 
+Every elasticsearch node maintains various networking clients and servers,
+protocols, and synchronous/asynchronous handling.
+
+## HTTP Transport
+
+Every node has an HTTP Transport Server (simply HTTP Transport) and it is a
+single entry point for all external clients (except cross-cluster). Management,
+ingestion, search, and everything else passes through HTTP server.
+
+Elasticsearch works over HTTP 1.1 and supports HTTP features such as: TLS,
+chunked transfer encoding, content compression, pipelining. Although we try to
+be HTTP spec compliant, Elastic is not a webserver. We support GET requests with
+payload (some old proxies might drop content), requests cannot be cached by
+middle boxes.
+
+By default HTTP server binds to the first available port in range
+9200-9300. There is no connections limit, but there is a limit on payload
+size. Default maximum payload is 100mb after compression. It's a very large
+number and almost never a good target that client should approach.
+
+Security is not enabled by default, meaning no TLS and authentication. These
+features are available in the x-pack/security module.
```

Comment:
I would not say "Security is not enabled by default" but instead just say "security features are achieved with separate x-pack modules". We had dedicated effort to ensure [security is on by default](https://github.com/elastic/elasticsearch/issues/80170) and basic security including authc/authz/TLS is in the free tier. It is rather disheartening to say the opposite.